diff --git a/kernel/sched/isolation.c b/kernel/sched/isolation.c
index 73386019efcb..c972885c776b 100644
--- a/kernel/sched/isolation.c
+++ b/kernel/sched/isolation.c
@@ -119,6 +119,14 @@ static int __init housekeeping_setup(char *str, unsigned long flags)
 		}
 	}
 
+	alloc_bootmem_cpumask_var(&non_housekeeping_mask);
+	if (cpulist_parse(str, non_housekeeping_mask) < 0) {
+		pr_warn("Housekeeping: nohz_full= or isolcpus= incorrect CPU range\n");
+		goto free_non_housekeeping_mask;
+	}
+
+	pr_info("isolcpus setup patch in effect\n");
+
 	if (!cpumask_subset(non_housekeeping_mask, cpu_possible_mask)) {
 		pr_info("housekeeping: kernel parameter 'nohz_full=' or 'isolcpus=' contains nonexistent CPUs.\n");
 		cpumask_and(non_housekeeping_mask, cpu_possible_mask,
@@ -131,12 +139,6 @@ static int __init housekeeping_setup(char *str, unsigned long flags)
 		return 0;
 	}
 
-	alloc_bootmem_cpumask_var(&non_housekeeping_mask);
-	if (cpulist_parse(str, non_housekeeping_mask) < 0) {
-		pr_warn("Housekeeping: nohz_full= or isolcpus= incorrect CPU range\n");
-		goto free_non_housekeeping_mask;
-	}
-
 	alloc_bootmem_cpumask_var(&housekeeping_staging);
 	cpumask_andnot(housekeeping_staging,
 		       cpu_possible_mask, non_housekeeping_mask);
