diff --git a/src/em_atomic_group.c b/src/em_atomic_group.c
index 4d7abc3..04f387a 100644
--- a/src/em_atomic_group.c
+++ b/src/em_atomic_group.c
@@ -168,6 +168,7 @@ ag_internal_enq(const atomic_group_elem_t *ag_elem, const em_event_t ev_tbl[],
 		const int num_events, const em_queue_prio_t priority)
 {
 	odp_event_t odp_ev_tbl[num_events];
+	(void) memset(odp_ev_tbl, 0, sizeof(odp_ev_tbl));
 	odp_queue_t plain_q;
 	int ret;
 
